odkx-src/injected-interfaces.rst

```diff
@@ -733,3 +786,520 @@ removeFirstQueuedAction
+The ``limit`` and ``offset`` parameters allow for pagination of the query results.
+The function serializes the SQL bind parameters to JSON and passes the query to the ``getOdkDataIf().arbitraryQuery`` method.
+Callback functions are registered to handle the success or failure of the arbitrary query.
+
+getRows
+~~~~~~~
+
+  **Parameters**:
+
+  - tableId: The ID of the table from which to retrieve rows.
+  - rowId: The ID of the specific row to retrieve. This can be ``null`` to retrieve all rows.
+  - successCallbackFn: A success callback function that is called when the retrieval of rows is successful.
+  - failureCallbackFn: A failure callback function that is called when there is an error in retrieving the rows.
+
+This function is used to retrieve rows from a specified table.
+You can either retrieve all rows from the table by passing ``null`` as the ``rowId``, or you can specify a particular row to retrieve.
+The function uses the ``getOdkDataIf().getRows`` method to fetch the rows.
+Callback functions are registered to handle the success or failure of the retrieval operation.
+
+getMostRecentRow
+~~~~~~~~~~~~~~~~~
+
+  **Parameters**:
+
+  - tableId: The ID of the table from which to retrieve the most recent row.
+  - rowId: The ID of the specific row to retrieve, typically the one you consider as the most recent. This can be ``null`` to retrieve the most recent row from the entire table.
+  - successCallbackFn: A success callback function that is called when the retrieval of the most recent row is successful.
+  - failureCallbackFn: A failure callback function that is called when there is an error in retrieving the most recent row.
+
+This function is used to retrieve the most recent row from a specified table.
+You can either specify a particular row as the most recent one, or you can retrieve the most recent row from the entire table by passing ``null`` as the rowId.
+The function uses the ``getOdkDataIf().getMostRecentRow`` method to fetch the most recent row.
+Callback functions are registered to handle the success or failure of the retrieval operation.
+
+changeAccessFilterOfRow
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+  **Parameters**:
+
+  - tableId: The ID of the table to which the row belongs.
+  - defaultAccess: The default access control for the row.
+  - rowOwner: The access control for the row owner.
+  - groupReadOnly: The access control for a group with read-only permission.
+  - groupModify: The access control for a group with modify permission.
+  - groupPrivileged: The access control for a privileged group.
+  - rowId: The ID of the specific row for which access control is to be changed.
+  - successCallbackFn: A success callback function that is called when the access control change operation is successful.
+  - failureCallbackFn: A failure callback function that is called when there is an error in changing the access control.
+
+This function is used to change the access control filter settings of a specific row in a table.
```

Comment:
May be helpful to add cross-link here to explaining access control filtering generally.